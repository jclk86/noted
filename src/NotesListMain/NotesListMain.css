.add-note-wrapper {
  width: 28%;
}

.add-note-btn {
  width: 100%;
  height: 50px;
  font-size: 16px;
  border-radius: 50px;
}
