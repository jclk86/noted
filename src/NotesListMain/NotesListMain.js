import React from "react";
import { getNotesForFolders } from "../helper-functions";
import Context from "../context";
import { NavLink, withRouter } from "react-router-dom";
import Note from "../Note/Note";
import "./NotesListMain.css";
import PropTypes from "prop-types";

class NotesListMain extends React.Component {
  static contextType = Context;

  render() {
    const notesForFolder = getNotesForFolders(
      this.context.notes,
      this.props.match.params.folderId
    );
    console.log(this.props);
    return (
      <ul>
        {notesForFolder.map(note => (
          <Note key={note.name} note={note} />
        ))}
        <div className="add-note-wrapper">
          <NavLink to="/add-note">
            <button className="add-note-btn">+ Add Note</button>
          </NavLink>
        </div>
      </ul>
    );
  }
}

NotesListMain.defaultProps = {
  notes: []
};

NotesListMain.propTypes = {
  folderId: PropTypes.string
};

export default withRouter(NotesListMain);
