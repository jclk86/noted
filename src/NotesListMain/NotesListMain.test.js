import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import NotesListMain from "./NotesListMain";
import Context from "../context";

describe(`NotesListMain component`, () => {
  it("renders a .NoteListMain by default", () => {
    const wrapper = shallow(<NotesListMain />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders a Note in ul for each notes in array", () => {
    const contextType = {
      notes: [
        {
          id: "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
          name: "Dogs",
          modified: "2019-01-03T00:00:00.000Z",
          folderId: "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
          content: "Corporis accusamus placeat.\n \rUnde."
        },
        {
          id: "d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1",
          name: "Cats",
          modified: "2018-08-15T23:00:00.000Z",
          folderId: "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
          content: "Eos\n \rlaudantium."
        },
        {
          id: "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1",
          name: "Pigs",
          modified: "2018-03-01T00:00:00.000Z",
          folderId: "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
          content: "Occaecati dignissimos\nvoluptatum nihil."
        },
        {
          id: "d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1",
          name: "Birds",
          modified: "2019-01-04T00:00:00.000Z",
          folderId: "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
          content: "Eum culpa odit."
        }
      ]
    };

    const match = {
      params: {
        folderId: "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1"
      }
    };

    const ul = shallow(
      <Context.Provider value={contextType}>
        <NotesListMain match={match} />
      </Context.Provider>
    ).find("ul");
    expect(toJson(ul)).toMatchSnapshot();
  });
});
