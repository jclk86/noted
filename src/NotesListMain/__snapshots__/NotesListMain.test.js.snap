// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesListMain component renders a .NoteListMain by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`NotesListMain component renders a Note in ul for each notes in array 1`] = `null`;
